api/v1alpha2/linodeobjectstoragekey_types.go

```diff
@@ -0,0 +1,130 @@
+)
+
+type BucketAccessRef struct {
+	BucketName  string `json:"bucketName"`
+	Permissions string `json:"permissions"`
+	Region      string `json:"region"`
+}
+
+// LinodeObjectStorageKeySpec defines the desired state of LinodeObjectStorageKey
+type LinodeObjectStorageKeySpec struct {
+	// BucketAccess is the list of object storage bucket labels which can be accessed using the key
+	// +kubebuilder:validation:MinItems=1
+	BucketAccess []BucketAccessRef `json:"bucketAccess"`
+
+	// CredentialsRef is a reference to a Secret that contains the credentials to use for generating access keys.
+	// If not supplied then the credentials of the controller will be used.
+	// +optional
+	CredentialsRef *corev1.SecretReference `json:"credentialsRef"`
+
+	// KeyGeneration may be modified to trigger a rotation of the access key.
+	// +kubebuilder:default=0
+	KeyGeneration int `json:"keyGeneration"`
+
+	// SecretType instructs the controller what type of secret to generate containing access key details.
+	// +kubebuilder:validation:Enum=Opaque;addons.cluster.x-k8s.io/resource-set
+	// +kubebuilder:default=Opaque
+	// +optional
+	SecretType corev1.SecretType `json:"secretType,omitempty"`
+}
+
+// LinodeObjectStorageKeyStatus defines the observed state of LinodeObjectStorageKey
+type LinodeObjectStorageKeyStatus struct {
+	// Ready denotes that the key has been provisioned.
+	// +optional
+	// +kubebuilder:default=false
+	Ready bool `json:"ready"`
+
+	// FailureMessage will be set in the event that there is a terminal problem
+	// reconciling the Object Storage Key and will contain a verbose string
+	// suitable for logging and human consumption.
+	// +optional
+	FailureMessage *string `json:"failureMessage,omitempty"`
+
+	// Conditions specify the service state of the LinodeObjectStorageKey.
+	// +optional
+	Conditions clusterv1.Conditions `json:"conditions,omitempty"`
+
+	// CreationTime specifies the creation timestamp for the secret.
+	// +optional
+	CreationTime *metav1.Time `json:"creationTime,omitempty"`
+
+	// LastKeyGeneration tracks the last known value of .spec.keyGeneration.
+	// +optional
+	LastKeyGeneration *int `json:"lastKeyGeneration,omitempty"`
+
+	// KeySecretName specifies the name of the Secret containing access key data.
+	// +optional
+	KeySecretName *string `json:"keySecretName,omitempty"`
+
+	// AccessKeyRef stores the ID for Object Storage key provisioned.
+	// +optional
+	AccessKeyRef *int `json:"accessKeyRef,omitempty"`
+}
+
+// +kubebuilder:object:root=true
+// +kubebuilder:resource:path=linodeobjectstoragekeys,scope=Namespaced,categories=cluster-api,shortName=lobjkey
+// +kubebuilder:subresource:status
+// +kubebuilder:metadata:labels="clusterctl.cluster.x-k8s.io/move-hierarchy=true"
+// +kubebuilder:printcolumn:name="ID",type="string",JSONPath=".status.accessKeyRef",description="The ID assigned to the access key"
+// +kubebuilder:printcolumn:name="Label",type="string",JSONPath=".metadata.name",description="The label of the access key"
```

Comment:
if `Label` and `Name` are a 1:1 mapping do we still need this separate field for label?